File: apps/ios/Orange Cloud/Orange Cloud/Core/Auth/OAuthConfig.swift
//
//  OAuthConfig.swift
//  Orange Cloud
//
//  在 Cloudflare Dashboard 创建 OAuth Client 后，填入真实 clientID。
//  redirect_uri 必须与 Dashboard 中注册的完全一致。
//

import Foundation

nonisolated enum OAuthConfig {
    /// 官方 OAuth Client（Cloudflare Dashboard → OAuth clients）。
    /// 仅供官方构建使用；自编译请自建 Client 与回调中转，见 CONTRIBUTING.md。
    static let clientID = "eae9090b8f240e6dd54d9926a55d56ce"

    /// 自定义 scheme，供 Web 后端 302 跳回 App
    static let callbackScheme = "orangecloud"

    // Cloudflare OAuth 只接受 https redirect_uri，指向 Web 后端回调中转（见 apps/web/README.md）
//    #if DEBUG
//    static let redirectURI = "http://localhost:3000/oauth/callback"
//    #else
    static let redirectURI = "https://orange-cloud.chatiro.app/oauth/callback"
//    #endif

    // Cloudflare OAuth 端点
    static let authorizationURL = URL(string: "https://dash.cloudflare.com/oauth2/auth")!
    static let tokenURL         = URL(string: "https://dash.cloudflare.com/oauth2/token")!
    static let revokeURL        = URL(string: "https://dash.cloudflare.com/oauth2/revoke")!
    static let userInfoURL      = URL(string: "https://dash.cloudflare.com/oauth2/userinfo")!

    // App 法律与支持链接
    static let privacyPolicyURL = URL(string: "https://orange-cloud.chatiro.app/privacy")!
    static let termsOfUseURL    = URL(string: "https://orange-cloud.chatiro.app/terms")!
}

File: apps/android/app/src/main/kotlin/jiamin/chen/orangecloud/core/auth/OAuthConfig.kt
package jiamin.chen.orangecloud.core.auth

import jiamin.chen.orangecloud.BuildConfig

/**
 * OAuth 配置（与 iOS Core/Auth/OAuthConfig.swift 对应）。
 * clientId 经 Gradle 按风味注入：play = 官方 Client，oss = 自编译者自填。
 * Cloudflare OAuth 只接受 https redirect，指向 Web 后端回调中转，再 302 跳回自定义 scheme。
 */
object OAuthConfig {
    /** 官方 Client 为 PKCE 公开客户端（非机密）；oss 风味默认空串。 */
    val clientId: String = BuildConfig.OAUTH_CLIENT_ID

    const val CALLBACK_SCHEME = "orangecloud"
    const val CALLBACK_HOST = "oauth"
    const val REDIRECT_URI = "https://orange-cloud.chatiro.app/oauth/callback"

    const val AUTHORIZATION_URL = "https://dash.cloudflare.com/oauth2/auth"
    /** 网页登出端点：添加账号时先登出再续跳授权页，避免复用上一个登录态（支持 ?to= 续跳）。 */
    const val LOGOUT_URL = "https://dash.cloudflare.com/logout"
    const val TOKEN_URL = "https://dash.cloudflare.com/oauth2/token"
    const val REVOKE_URL = "https://dash.cloudflare.com/oauth2/revoke"
    const val USERINFO_URL = "https://dash.cloudflare.com/oauth2/userinfo"
}
